tests/ui_tools/test_boxes.py

```diff
@@ -242,6 +242,49 @@ def test_generic_autocomplete_mentions_subscribers(self, write_box, text,
         typeahead_string = write_box.generic_autocomplete(text, state)
         assert typeahead_string == required_typeahead
 
+    @pytest.mark.parametrize('text', [
+        '@H', '@Hu', '@Hum', '@Huma', '@Human',
+        '@**', '@**H', '@**Hu', '@**Hum', '@**Huma', '@**Human'
+    ])
+    @pytest.mark.parametrize('matching_users, distinct_matching_users', [
+        (['Human Myself', 'Human 1', 'Human 2', 'Human Duplicate',
+          'Human Duplicate'],
+         ['@**Human Myself**', '@**Human 1**', '@**Human 2**',
+          '@**Human Duplicate|13**', '@**Human Duplicate|14**'])
+    ])
+    def test_generic_autocomplete_user_mentions(self, write_box, text, mocker,
+                                                distinct_matching_users,
+                                                matching_users, state=1):
+        _process_typeaheads = mocker.patch(BOXES
+                                           + '.WriteBox._process_typeaheads')
+
+        write_box.generic_autocomplete(text, state)
+
+        _process_typeaheads.assert_called_once_with(distinct_matching_users,
+                                                    state, matching_users)
+
+    @pytest.mark.parametrize('text', [
+        '@_', '@_H', '@_Hu', '@_Hum', '@_Huma', '@_Human'
```

Comment:
Can we combine with the previous test?

Reply:
The `distinct_matching_users` list differs for these autocomplete `text`'s, which is why I moved these to another test.

Reply:
I meant that the typeahead would need `@_**Human 1**` instead of `@**Human 1**` because this test is for silent mentions, which makes the `distinct_matching_users` list different.